plugins/inventory/cobbler.py

```diff
@@ -145,7 +145,7 @@ def _get_connection(self):
             self.connection = xmlrpc_client.Server(self.cobbler_url, allow_none=True)
             self.token = None
             if self.get_option('user') is not None:
-                self.token = self.connection.login(self.get_option('user'), self.get_option('password'))
+                self.token = self.connection.login(str(self.get_option('user')), str(self.get_option('password')))
```

Comment:
```suggestion
                self.token = self.connection.login(to_text(self.get_option('user')), to_text(self.get_option('password')))
```

Reply:
Please excuse me if my lack of knowledge is getting in a way, but shouldn't I see "string" type after "to\_text()" if that is what you do?  
just added plain print to that plugin to verify:  
`print("str object type:{}, to_text object type: {}".format(type(str(self.get_option('user'))), type(to_text(self.get_option('password')))))`

and I get 

`str object type:<class 'str'>, to_text object type: <class 'ansible.parsing.yaml.objects.AnsibleUnicode'>`  
so at the end, we have changed nothing, we are still forwading AnsibleUnicode to xmlrpc client.

Reply:
Hmm, in that case it seems that `AnsibleUnicode` is too clever for `to_text` to properly work. Please use
```
from ansible.module_utils.six import text_type

... text_type(get_option('user')) ... text_type(get_option('password')) ...
```
instead. That produces a real `str` on Python 3, and a real `unicode` on Python 2.